java/readme.md

```diff
@@ -0,0 +1,213 @@
+
+```java title="Main.java"
+package org.example;
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+
+import java.util.concurrent.ExecutionException;
+
+public class Producer {
+    public static void main(String[] args) {
+        produce("demo-topic");
+    }
+
+    public static void produce(String topicName) {
+        try (org.apache.kafka.clients.producer.Producer<Void, String> producer = new KafkaProducer<>(Main.clientProperties())) {
+            for (var i = 0; i < 100; i++) {
+                ProducerRecord<Void, String> record = new ProducerRecord<>(topicName, String.format("asynchronous message #%d", i));
+                producer.send(record, (r, e) -> {
+                    if (e == null) {
+                        System.out.printf("Sent to topic '%s' at offset %d\n", r.topic(), r.offset());
+                    } else {
+                        System.out.println("Error sending message: " + e);
+                    }
+                });
+
+            }
+        }
+    }
+}
+```
+
+
+## Create a consumer to read data from the topic
+Create a file named `src/main/java/org/example/Consumer.java` and paste the code below. Note that the consumer requires slightly more configuration than the producer.
+
+```java title="Consumer.java"
+package org.example;
+
+import org.apache.kafka.clients.consumer.*;
+import org.apache.kafka.common.TopicPartition;
+
+import java.time.Duration;
+import java.util.*;
+
+public class Consumer {
+    public static void main(String[] args) {
+        consume("demo-topic");
+    }
+
+    public static void consume(String topicName) {
+        var props = Main.clientProperties();
+        props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+        try (org.apache.kafka.clients.consumer.Consumer<String, String> consumer = new KafkaConsumer<>(props)) {
+            var topicPartitions = Collections.singleton(new TopicPartition(topicName, 0));
+            consumer.assign(topicPartitions);
+            consumer.seekToBeginning(topicPartitions);
+            while (true) {
+                ConsumerRecords<String, String> records = consumer.poll(Duration.ofSeconds(5));
+                if (records.count() == 0) break;
+                for (ConsumerRecord<String, String> receivedRecord : records) {
+                    System.out.printf("topic: %s (%d|%d), key: %s, %s\n",
+                            receivedRecord.topic(), receivedRecord.partition(), receivedRecord.offset(),
+                            receivedRecord.key(), receivedRecord.value());
+                }
+            }
+        }
+    }
+}
+```
+
+
+## Build and run
+
+```bash
+# Produce the executable jar file
+mvn package
+# Create the topic
+java -jar target/redpanda-client-1.0-SNAPSHOT-jar-with-dependencies.jar
+# Produce some data
+java -cp target/redpanda-client-1.0-SNAPSHOT-jar-with-dependencies.jar org.example.Producer
```

Comment:
The SLF4J log is slightly anoying. I guess it is the default of the Kafka client. Since we own the config in the pom file/code, can we get rid of it?

```
$ java -cp target/redpanda-client-1.0-SNAPSHOT-jar-with-dependencies.jar org.example.Producer
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Sent to topic 'demo-topic' at offset 0
...
```